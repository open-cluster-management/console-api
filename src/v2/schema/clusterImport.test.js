/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('Cluster Import Resolver', () => {
  test('Create Kubernetes Cluster Resource for Import', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
            createClusterResource(body: "{\\"clusterName\\":\\"foo\\", \\"clusterNamespace\\":\\"foo\\", \\"clusterLabels\\": {}}")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Create Kubernetes Cluster Resource for Import should return error msg', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
            createClusterResource(body: "")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });
});
