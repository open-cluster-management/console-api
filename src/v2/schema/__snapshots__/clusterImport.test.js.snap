// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cluster Import Resolver Create Kubernetes Cluster Resource for Import 1`] = `
Object {
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 35,
          "line": 3,
        },
      ],
      "message": "Unknown argument \\"body\\" on field \\"createClusterResource\\" of type \\"Mutation\\".",
    },
    Object {
      "locations": Array [
        Object {
          "column": 13,
          "line": 3,
        },
      ],
      "message": "Field \\"createClusterResource\\" argument \\"cluster\\" of type \\"JSON!\\" is required but not provided.",
    },
  ],
}
`;

exports[`Cluster Import Resolver Create Kubernetes Cluster Resource for Import should return error msg 1`] = `
Object {
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 35,
          "line": 3,
        },
      ],
      "message": "Unknown argument \\"body\\" on field \\"createClusterResource\\" of type \\"Mutation\\".",
    },
    Object {
      "locations": Array [
        Object {
          "column": 13,
          "line": 3,
        },
      ],
      "message": "Field \\"createClusterResource\\" argument \\"cluster\\" of type \\"JSON!\\" is required but not provided.",
    },
  ],
}
`;
